baybe/utils/plotting.py

```diff
@@ -0,0 +1,78 @@
+import pandas as pd
+import seaborn as sns
+
+
+def create_plots(data: pd.DataFrame, path: Path, base_name: str, **kwargs) -> None:
+    """Create plots from a given data frame and save them as a svg file.
+
+    The plots will be saved in the location specified by ``path``.
+    The attribute ``base_name`` is used to define the name of the outputs.
+
+    If the ``SMOKE_TEST`` variable is set, no plots are being created and this method
+    immediately returns.
+
+    Using the ``BAYBE_MULTIVERSION_PLOTS`` environment variable, it is possible to
+    create plots for the light and dark version of the documentation. If this variable
+    is not set, a single file named ``{base_name}_check.svg`` is created.
+
+    Args:
+        data: The data frame containing the data to be plotted.
+        path: The path to the directory in which the plots should be saved.
+        base_name: The base name that is used for naming the output files.
+        **kwargs: Keyword arguments. They are directly passed to ``sns.lineplot`` and
+            are used for specifying the plot.
+    """
+    # If we run a SMOKE_TEST, we immediately return
+    if "SMOKE_TEST" in os.environ:
+        return
+    # We see if we happen to find the plotting themes in the current folder.
+    # This is e.g. the case if we convert the file to a jupyter notebook.
+    try:
+        themes = json.load(open("plotting_themes.json"))
+    except FileNotFoundError:
+        # Try to find the plotting themes by backtracking
+        # Get the absolute path of the current script
+        script_path = Path(sys.argv[0]).resolve()
+        # Backtrack until either the "baybe" folder is found or backtracking is no
+        # longer possible
+        while not script_path.name == "baybe" and script_path != script_path.parent:
+            script_path = script_path.parent
+        if script_path == script_path.parent:
+            warnings.warn("No themes for plotting found. A default theme is used.")
+            themes = {"check": {"font_scale": 1.75, "rc_params": {}}}
+        else:
+            # Open the file containing all the themes
+            # This currently still assumes that the file is in the repo folder
+            themes = json.load(open(script_path / "plotting_themes.json"))
+
+    # Environment variables for checking whether we want to have multiversion plots
+    BAYBE_MULTIVERSION_PLOTS = "BAYBE_MULTIVERSION_PLOTS" in os.environ
+
+    # Choose either all available or just the `check` theme.
+    chosen_themes = themes if BAYBE_MULTIVERSION_PLOTS else ["check"]
+    for theme_name in chosen_themes:
+        # Extract and set the current theme
+        theme = themes[theme_name]
+        font_scale, rc_params = theme["font_scale"], theme["rc_params"]
+        sns.set_theme(style="ticks", font_scale=font_scale, rc=rc_params)
+        # Only if kwargs are being provided, a plot is actually created
+        if kwargs:
+            sns.lineplot(data=data, **kwargs)
```

Comment:
why did you decide to do that plot here? in the last call I remember the possibility of passing `fig` object to this utlity instead of the dataframe was prefered

in the examples then we could use this design pattern:
```python
fig, ax = plt.subplots()
sns.lineplot(..., ax=ax)

create_example_plots(fig, ...)

if not SMOKE_TEST:
    plt.show()
```
(you might need to turn off the immedaite showing of the figure by linplot via a `plt.` command i dont remember)

Reply:
The problem is that I did not manage to create multiple plots when providing it with a figure directly. No matter what I did and tried, it always only used the first theme that I provided, saved the figure as a file, but created empty files after this. I tried a lot of different things, including deepcopies of the fig object, re-setting the active figure and so on and nothing worked. Thus, Adrian and I agreed on "Ok let's just create the figures for now and discuss at a later point". 
So if you have any idea or solution how to get the multi-version plot running with just a single figure being provided, I'd be more than happy to implement it :)

Reply:
I do not know what exactly I tried, but I can re-iterate and post some screenshots here

Reply:
First variant:
<img width="529" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/db457beb-d91e-4c76-a2cd-6cbbf5c97e5c">
Here, `figure` is created in the example and the given to the function. This only creates a single plot, namely for the `dark`theme as this is the first one. If I use `plt.close()` in the end instead of `plt.clf()`, then I get three times the same plot.

Reply:
you using the same deepcopy three times and not three deepcopies

Reply:
And upon further investigating this, it seems that it is not possible to change the style after a figure has been created if I undestand https://stackoverflow.com/questions/35127724/how-to-set-seaborn-styles-on-existing-matplotlib-axes correctly:
![image](https://github.com/emdgroup/baybe/assets/84016733/faa4127f-2145-4ac5-9451-a5642f72fd7b)
![image](https://github.com/emdgroup/baybe/assets/84016733/456f9286-259e-4c2c-9621-a6555e49801c)

Reply:
Regarding the use of the same deepcopy three times: But this is not the case in my very last screenshot, and as I wrote, I also do not get the expected behavior when I use `plt.figure(deepcopy(figure))` which should definitely create new objects.

Reply:
I have now finally changed this to figure, so feel free to resolve this if this is gucci.